### PR TITLE
Add SW Version, Model No, and Serial No. Datapoints for ODUs

### PR DESCRIPTION
**Pull request by justbruce**
Add request_mod_address 7 as integral to this change

### CODE REVIEW

**Comment by justbruce** [comment]
Should close #277

**Review by barndawgie — DISMISSED** [review]

**Comment by barndawgie** [comment]
Also, @justbruce, please see the README for details on installing and enabling pre-commit - that will catch things like whitespace issues locally.

**Comment by justbruce** [comment]
> Also, @justbruce, please see the README for details on installing and enabling pre-commit - that will catch things like whitespace issues locally.

I'm aware - however, I'm not a software developer.  I do a little fiddling here and there, when I have time (you might note many of my interactions happen after midnight.) 

I use the web client for it's convenience, and unfortunately it hides whitespace sometimes.

**Comment by barndawgie** [comment]
> > Also, @justbruce, please see the README for details on installing and enabling pre-commit - that will catch things like whitespace issues locally.
> 
> I'm aware - however, I'm not a software developer. I do a little fiddling here and there, when I have time (you might note many of my interactions happen after midnight.)
> 
> I use the web client for it's convenience, and unfortunately it hides whitespace sometimes.

No problem. Please take a look at my feedback when you have some time :-)

**Review by justbruce — COMMENTED** [review]
Requested substitutions & Icon info

**Review by justbruce — COMMENTED** [review]
whitespace

File: econet_hvac_odu.yaml [hunk 1]
```diff
@@ -244,3 +251,31 @@ sensor:
...
+    request_mod: 7
+    src_address: ${odu_address}
+    icon: "mdi:information-box"
+    entity_category: "diagnostic"
+  - platform: econet
+    name: "ODU Software Version"
+    id: odu_sw_version
+    sensor_datapoint: ODUSWVER
+    request_once: true
+    src_address: ${thermostat_address}
```
Comment (barndawgie) [inline econet_hvac_odu.yaml:279]: Is this correct? This datapoint comes from the Thermostat rather than the ODU?

**Review by barndawgie — APPROVED** [review]
LGTM assuming that you've tested and it is working for you.

File: econet_hvac_odu.yaml (hunk 1 above)
Comment (justbruce) [reply econet_hvac_odu.yaml:279]: Yes.

  I initially tried just requesting from the ODU. [1024 decimal, 0x400 hex]. But either our implementation has something funky, or there is a bug in the Rheem econet implementation - when I would request from just the ODU, the thermostat AND the furnace would also reply, and which order they replied in would determine the value displayed.

  So, given that this datapoint is available, I dropped it in as it got the job done.

**Comment by justbruce** [comment]
> LGTM assuming that you've tested and it is working for you.

Tested on my unit, working for the past 6 hours just fine.

**Pull request merged**